transifex/push.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+"""
+This script can be used to automatically extract translation strings,
+commit, push, and merge them to their respective repos, and then push them to transifex.
+
+To use, export an environment variable `GITHUB_ACCESS_TOKEN`. The token requires
+GitHub's "repo" scope.
+
+Run the script from the root of this repo.
+
+    python3 transifex/push.py git@github.com:edx/course-discovery.git
+
+If you want to use a custom merge method pass the --merge-method option.
+
+    python3 transifex/push.py git@github.com:edx/course-discovery.git --merge-method rebase
+"""
+from argparse import ArgumentParser
+from utils.common import MERGE_METHODS, logger, repo_context
+
+
+# The name of the branch to use.
+BRANCH_NAME = 'update-translation-strings'
+
+# The commit message to use.
+MESSAGE = 'Update translation strings'
+
+
+def push(clone_url, merge_method=None):
+    """Extracts translations for the given repo, commits them, pushes them to GitHub, opens a PR, waits for status
```

Comment:
Since the marketing site repository is private, we cannot easily/reliably setup automatic source translation updates on Transifex. Instead, we plan to use this job to manually push updates to Transifex on a regular schedule. I do not see any benefit to requiring devs to extract_translations before every merge, as the updated strings would not make it to Transifex until this job is run anyway. Given that, I don't think it makes sense to add an extra step to the marketing site development process when we can add one extra line to the job to do the extraction just prior to pushing to Transifex.